pkg/importer/vddk-datasource_amd64.go

```diff
@@ -340,7 +340,9 @@ func (vmware *VMwareClient) FindDiskInSnapshotTree(snapshots []types.VirtualMach
 		if disk := vmware.FindDiskInSnapshot(snapshot.Snapshot, fileName); disk != nil {
 			return disk
 		}
-		return vmware.FindDiskInSnapshotTree(snapshot.ChildSnapshotList, fileName)
+		if disk := vmware.FindDiskInSnapshotTree(snapshot.ChildSnapshotList, fileName); disk != nil {
```

Comment:
What was the linter complaining about here? or is this just about consistency?

Reply:
@akalenyu - The linter warning was:
```
SA4004: the surrounding loop is unconditionally terminated (staticcheck)
```

And I think it's an actual bug, because before this change, the function only search for the disk in the first snapshot's ChildSnapshotList, and exited anyway, even if not found. The original version never search in more than one snapshot in the slice. 

As I'm not familiar with the domain, and the slice is always with length of 1, then maybe it would better to rewrite this function. But for now, just by reading golang code - this is a bug.

Reply:
> before this change, the function only search for the disk in the first snapshot's ChildSnapshotList

Isn't it fine since it's recursive?

Reply:
Let's say you have 3 snapshots in the slice, each with 3 children. And let's say that the required disk is the third child of the third snapshot.

The function will start checking the first snapshot. The first condition will be false so it will start checking the children. First the first child, and the first condition is false again, then it returns nil, since the first child does not have children. But since it's a `return`, the rest of the children are not checked. now we again at the original call. And again, since it's a call with `return`, the 2nd and the 3rd snapshot are not checked.